### PR TITLE
Add a ship command

### PR DESCRIPTION
**Pull request by duncte123**

### CODE REVIEW

File: utils/finderUtil.js [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    singleMatch = resolver.resolve(userId);
+  } else if (USER_TAG.test(query)) {
+    // The first comma ignores the first element
+    // Read more: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Ignoring_some_returned_values
+    const [, username, discriminator] = USER_TAG.exec(query);
+
+    // default to null as find may return undefined
+    singleMatch =
+      resolver.cache.find(
+        u => u.username === username && u.discriminator === discriminator
```
Comment (duncte123) [inline utils/finderUtil.js]: This will break due to u being a member, needs a simple check to fix

**Comment by iamdeja** [comment]
> 
> 
> I would do a check on if the file exists instead

`exists` is deprecated however, and the error handling is part of the node [recommendation](https://nodejs.org/api/fs.html#fs_fs_exists_path_callback)

File: commands/Ship.js [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+    }
+
+    return false;
+  }
+
+  findMessage(score) {
+    return messages.find(
+      obj => obj.min <= score && (obj.max ? score <= obj.max : true)
+    ).message;
+  }
```
Comment (Leguu) [inline commands/Ship.js:51]: Potential simplification to:
```js
  findMessage(score) {
    for (message in messages) {
      if (score <= message.max) return message.message;
    }
  }
```
You can remove all the `min` values in ship_messages.json. Don't forget to add a `max` of 100 in the last one.

File: commands/Ship.js [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+    for (const ids of this.#riggedPairs) {
+      if (
+        (ids[0] === id1 && ids[1] === id2) ||
+        (ids[0] === id2 && ids[1] === id1)
+      ) {
+        return true;
+      }
+    }
+
+    return false;
```
Comment (Leguu) [inline commands/Ship.js:44]: Do
```js
return this.#riggedPairs.some(ids => {
  return (ids[0] === id1 && ids[1] === id2) ||
         (ids[0] === id2 && ids[1] === id1);
});
```

File: commands/Ship.js [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+      return user;
+    }
+
+    return null;
+  }
+
+  // Escape markdown characters from names
+  escapeName(user) {
+    return user.displayName.replace(/[`*_~>]/g, "\\$&");
+  }
```
[the remark this answers is not in the document]
Comment (iamdeja) [reply commands/Ship.js:100]: I second having that one as an util. Just rewrote it for Tod command

File: commands/Ship.js (hunk 2 above)
Comment (iamdeja) [reply commands/Ship.js:51]: For good practice will need to add `obj.hasOwnProperty(prop)` before the if clause if that's the route to go

**Pull request merged**